src/DataViews/LiveFunctionsDataViewTest.cpp

```diff
@@ -800,4 +820,35 @@ TEST_F(LiveFunctionsDataViewTest, ColumnSortingShowsRightResults) {
     // Sort by descending
     { sort_and_verify(column, orbit_data_views::DataView::SortingOrder::kDescending); }
   }
+}
+
+TEST_F(LiveFunctionsDataViewTest, OnDataChangeResetsHistogram) {
+  EXPECT_CALL(app_, ShowHistogram(nullptr, "", orbit_grpc_protos::kInvalidFunctionId)).Times(1);
+
+  view_.OnDataChanged();
+}
+
+TEST_F(LiveFunctionsDataViewTest, OnRefreshWithNoIndicesResetsHistogram) {
+  EXPECT_CALL(app_, ShowHistogram(nullptr, "", orbit_grpc_protos::kInvalidFunctionId)).Times(2);
+
+  view_.OnRefresh({}, RefreshMode::kOnFilter);
+  view_.OnRefresh({}, RefreshMode::kOther);
+}
+
+MATCHER(ExpectedDurations, "") {
+  auto durations = *arg;
+  return durations == GetDurationsForInstrumentedFunctions();
+}
```

Comment:
You could also consider using `Pointee(Eq(GetDurationsForInstrumentedFunctions())` instead of a custom matcher.